### DISCUSSION TITLE
Custom function for mean prior

### DISCUSSION THREAD

**Discussion by Jahnvi99** [cornellius-gp/gpytorch#1669] [Q&A]
How to define a negative quadratic function as prior for the mean?

**Comment by gpleiss** [comment]
```python
class QuadraticMean(gpytorch.means.Mean):
    def __init__(self):
        # Define parameters for your quadratic
        self.a = -2.  # could be a learnable parameter

    def forward(self, x):
        return self.a * x.norm(dim=-1)
```

**Reply by Jahnvi99** [reply]
Thank you so much for providing me with the answer! But i am getting this error, using QuadraticMean() in the [simple regression example of gyptorch](https://docs.gpytorch.ai/en/latest/examples/01_Exact_GPs/Simple_GP_Regression.html) 
![Capture](https://user-images.githubusercontent.com/46685060/123416151-d1d2c900-d5d3-11eb-82e1-d258ee74bb47.JPG)
.

**Reply by gpleiss** [reply]
Can you include a reproducible example of your error?

**Reply by lukasfro** [reply]
For anyone that stumbles on this thread...

I got the same error message: `'QuadraticMean' object has no attribute 'training'`. 
What helped for me was to call the base class's `__init__` method.
The following mean function works just fine for me:

```python
class QuadraticMean(gpytorch.means.Mean):
    def __init__(self):        
        super(QuadraticMean, self).__init__()
        self.a = -2.0 

    def forward(self, x):
        return self.a * x.norm(dim=-1)
```

**Reply by gpleiss** [reply]
Oops, yes thanks for catching this @lukasfro !
